js/foam/u2/PropertyView.js

```diff
@@ -26,36 +26,51 @@ CLASS({
     'foam.u2.Input'
   ],
 
-  imports: [ 'data' ],
+  imports: [ 'data$' ],
 
   properties: [
     {
+      name: 'data',
+      postSet: function(old, nu) {
+        this.bindData_(old, nu);
+      },
+    },
+    {
       name: 'prop'
     },
     {
       name: 'view'
     },
+    {
+      name: 'child_',
+    },
     [ 'nodeName', 'tr' ]
   ],
 
   methods: [
-    function init() {
-      this.SUPER();
-
+    function initE() {
       var view = this.view || this.prop.toPropertyE();
       var prop = this.prop;
 
       // TODO: remove check once all views extend View
       view.fromProperty && view.fromProperty(prop);
 
-      // TODO: Why can't I just define this with data$ in the view constructor?
-      // Is data$ linking the wrong way?
-      Events.link(this.data.propertyValue(prop.name), view.data$);
-
-      this.initE(view);
+      this.child_ = view;
+      this.buildE(view);
+      this.bindData_(null, this.data);
     },
-    function initE(view) {
+    function buildE() {
```

Comment:
This is a misleading name since it isn't building an Element, just configuring one.
Could you move the code in initE() to init() and then rename this initE()?